delete.js

```diff
@@ -1,35 +1,42 @@
-const deleter = {
-  name_colunm_index: 0,
+class Deleter {
+  constructor(mainObject) {
+    this.name_colunm_index = 0;
+    this.allContacts = mainObject.addressBook;
+  }
 
   findContactToDelete(nameInput) {
-    let rows = addressBook.buildContactList();
+    let rows = this.allContacts.buildContactList();
     return rows.find(row => row.cells[this.name_colunm_index].textContent === nameInput);
-  },
+  }
 
   deleteContact(toBeDeleted) {
     toBeDeleted.remove();
   }
-};
+}
 
-const deletionUi = {
-  _userEntry: document.getElementById('user_contact_name'),
+class DeletionUi {
+  constructor(deleterObject, mainObject) {
+    this._userEntry = document.getElementById('user_contact_name');
+    this.eraser = deleterObject;
+    this.outputer = mainObject.mainUi;
+  }
 
   get userEntry() {
     return this._userEntry.value;
-  },
+  }
 
   _deletionProcedure() {
-    let addressToDelete = deleter.findContactToDelete(this.userEntry);
+    let addressToDelete = this.eraser.findContactToDelete(this.userEntry);
 
     if(!addressToDelete){
-      mainUi.displayInfoMessage = 'That contact does not exist';
+      this.outputer.displayInfoMessage = 'That contact does not exist';
       return;
     }else {
-      mainUi.displayInfoMessage = '';
```

Comment:
Since you have `return` inside of the previous `if`, you don't need this `else` anymore, as it will never be reached, and you can simplify the code like this (Re-read [what we saw in the previous PR](https://github.com/Matt-Warnock/address_book/pull/4#discussion_r423288334)):

```js
    if (!addressToDelete) {
      this.outputer.displayInfoMessage = 'That contact does not exist';
      return;
    }

    this.outputer.displayInfoMessage = '';
    this.eraser.deleteContact(addressToDelete);
```